src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -175,6 +179,28 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
             globalConfig = config.Globals;
             config.OnConfigChange("Defaults", () => nodeConfig = config.GetOrCreateNodeConfigurationForSilo(name));
 
+            // Configure DI using Startup type
+            bool usingCustomServiceProvider;
+            Action<IServiceCollection> configureServices = serviceCollection =>
+            {
+                serviceCollection.AddSingleton(this);
```

Comment:
Point taken. The container should never be used before it's been constructed and it's never actually accessed (only referenced) in the constructor, the lines below this are just declaring that the `IMembershipOracle` implementation requires a `Silo`.

I plan to make just minimal changes to use the container in this PR and intend to move more into the container in future PRs.
It would be best if the container itself was built outside of the `Silo` and the silo is constructed by it.